src/main/scala/play/dev/filewatch/PollingFileWatchService.scala

```diff
@@ -41,13 +41,21 @@ class PollingFileWatchService(val pollDelayMillis: Int) extends FileWatchService
 object SourceModificationWatch {
   type PathFinder = () => Iterator[ScalaFile]
 
+  private def listFiles(sourcesFinder: PathFinder): Set[ScalaFile] = sourcesFinder().toSet
+
+  private def findLastModifiedTime(files: Set[ScalaFile]): Long = {
+    if (files.nonEmpty) files.maxBy(_.lastModifiedTime).lastModifiedTime.toEpochMilli
+    else 0L
+  }
+
   @tailrec def watch(sourcesFinder: PathFinder, pollDelayMillis: Int, state: WatchState)(terminationCondition: => Boolean): (Boolean, WatchState) =
```

Comment:
`sourcesFinder` is [defined here](https://github.com/playframework/play-file-watch/blob/1.1.5/src/main/scala/play/dev/filewatch/PollingFileWatchService.scala#L22), listing the files recursively. A possible problem (still need to investigate) is that apparently, better-files is not closing the [`DirectoryStream`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/DirectoryStream.html) as suggested by `walk` docs:

> The returned stream encapsulates one or more DirectoryStreams. If timely disposal of file system resources is required, the try-with-resources construct should be used to ensure that the stream's close method is invoked after the stream operations are completed. Operating on a closed stream will result in an IllegalStateException.

See better files code here: https://github.com/pathikrit/better-files/blob/v2.17.1/core/src/main/scala/better/files/File.scala#L457-L467

This could lead to a bug similar to https://github.com/playframework/playframework/pull/7671.

Reply:
Forget this. The close is happening here:

https://github.com/pathikrit/better-files/blob/v2.17.1/core/src/main/scala/better/files/Implicits.scala#L259-L260

:-)